spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -93,6 +106,16 @@ public MessageChannel resolveDestination(String channelName) {
 					channel = this.bindingTargetFactory.createOutput(channelName);
 					this.beanFactory.registerSingleton(channelName, channel);
 					channel = (MessageChannel) this.beanFactory.initializeBean(channel, channelName);
+					if (this.newBindingCallback != null) {
+						ProducerProperties producerProperties = this.bindingService.getBindingServiceProperties()
+								.getProducerProperties(channelName);
+						Object extendedProducerProperties =
+								this.bindingService.getExtendedProducerProperties(channel, channelName);
+						this.newBindingCallback.configure(channelName, channel, producerProperties,
```

Comment:
Maybe we could only call the `callback.configure` only when the `extendedProducerProperties` isn't null ?

Reply:
Why would we not want to configure just the common properties if the binder does not support extended properties? Perhaps the callback javadocs should clarify that the extension might be null.